[core] Cover cpplint for `/src/ray/object_manager` (excluding `plasma`)

elimelt opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

As part of Run cpplint on all `.cc`/`.h` files in `src/ray/object_manager` and `src/ray/object_manager/test` 

## Related issue number

Main issue: #50583
Closes sub issue: #51084

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a
           method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

dentiny reviewed (commented):
Thanks for the effort! Really appreciate it; meanwhile could you please update the precommit hook: https://github.com/ray-project/ray/blob/master/.pre-commit-config.yaml?

elimelt commented:
Sorry everyone, rebasing error!

elimelt commented:
@dentiny I fixed the issue but I'm not able to un-request reviews. Would you or any maintainer be able to do so for me? Sorry again about this, I hadn't signed off on a commit so I dropped it but also rewrote history for all the branches in master while doing so 🤦‍♂️

dentiny commented on .pre-commit-config.yaml:
```diff
@@ -47,7 +47,7 @@ repos:
       - id: cpplint
         args: ["--filter=-whitespace/braces,-whitespace/line_length,-build/c++11,-build/c++14,-build/c++17,-readability/braces,-whitespace/indent_namespace,-runtime/int,-runtime/references,-build/include_order"]
-        files: ^src/ray/(util|raylet_client|internal|scheduling|pubsub|object_manager/plasma|rpc(?:/.*)?)/.*\.(h|cc)$
+        files: ^src/ray/(util|raylet_client|internal|scheduling|pubsub|object_manager|rpc(?:/.*)?)/.*\.(h|cc)$
```
I'm wondering why do we remove `object_manager` here?

elimelt replied:
`/plasma` changes were already done, which is a subdirectory of `object_manager`, the code I linted and fixed.

dentiny replied:
ohohoh sorry I read it wrong!

dentiny commented:
> Would you or any maintainer be able to do so for me?

No worries! Just started CI for you, feel free to ping me when it passes!

dentiny commented:
@aslonnie / @jjyao / @edoakes Could you please help merge the PR? Thank you!

dentiny reviewed (approved):
LGTM, thank you!

dentiny commented on src/ray/object_manager/test/get_request_queue_test.cc:
```diff
@@ -14,18 +14,31 @@
 
-using namespace ray;
-using namespace testing;
+// Removed: using namespace ray;
```
Oh could you please cleanup the comment?

elimelt replied:
Done!

dentiny commented:
@edoakes / @aslonnie / @jjyao could you please take a look and help merge? Thank you!

elimelt commented:
@dentiny Also happy to close this PR and open another if it's a hassle to remove the irrelevant review requests. let me know if this is preferable
